src/shogun/mathematics/RandomNamespace.h

```diff
@@ -0,0 +1,122 @@
+#ifndef __RANDOM_NAMESPACE_H__
+#define __RANDOM_NAMESPACE_H__
+
+#include <shogun/lib/common.h>
+#include <shogun/lib/config.h>
+
+#include <algorithm>
+#include <random>
+
+namespace shogun
+{
+	namespace random
+	{
+		/** Reorders the elements in [first,  last) randomly
+		 *
+		 * @param first an iterator to the first element in the range
+		 * @param last an iterator to the last element in the range
+		 */
+		template <typename RandomIt, typename PRNG>
+		static inline void shuffle(RandomIt first, RandomIt last, PRNG& prng)
```

Comment:
why do we need this wrapper for std::shuffle?

Reply:
btw do you think it would be worth to set PRNG as a forwarding reference i.e. just in case of std::shuffle? `PRNG&&`

Reply:
PRNG should always be an l-value, so std::shuffle wouldn't use move semantics, but would accept it as an lvalue reference

Reply:
but using `PRNG&&` covers both cases (lvalue reference and forwarding reference), so it's better

Reply:
ok so then we change it? :)